classes/Address.php

```diff
@@ -520,8 +520,9 @@ public static function aliasExist($alias, $id_address, $id_customer)
     public function getFieldsRequiredDB()
     {
         $this->cacheFieldsRequiredDatabase(false);
-        if (isset(self::$fieldsRequiredDatabase['Address'])) {
-            return self::$fieldsRequiredDatabase['Address'];
+        $objectName = $this->getObjectName();
+        if (isset(self::$fieldsRequiredDatabase[$objectName])) {
+            return self::$fieldsRequiredDatabase[$objectName];
```

Comment:
`ObjectModel::getFieldsRequiredDB()` already exists and has a different behavior : it looks for fields in DB (`ps_required_field` table).
`Address::getFieldsRequiredDB()` tho, looks for cached required fields only. It never looks into DB.
Required fields seem to be cached no matter what. So this is functionally ok. But maybe we should clean this up... What do you think ?